src/Database/Expression/QueryExpression.php

```
@@ -342,9 +342,21 @@ public function in($field, $values, $type = null)
      */
     public function addCase($conditions, $values = [], $types = [])
     {
+        deprecationWarning('QueryExpression::addCase() is deprecated, use case() instead.');
+
         return $this->add(new CaseExpression($conditions, $values, $types));
     }
 
+    /**
+     * Returns a new case expression object.
+     *
+     * @return \Cake\Database\Expression\CaseExpressionInterface
+     */
+    public function case(): CaseExpressionInterface
```

Is there a reason the `value()` parameter isn't supported here?

---

Not really... I just grew accustom to the `value()` method (which once replaced a really awkward constructor syntax), and when I ported this over to the core, I just didn't think of it 🤷‍♂️

Would you like me to add support for it?

---

I think it would read more like sql with it unless you have technical limits here.

---

Thinking about it... `null` would technically be a problem here, as we cannot distinguish between no value passed and explicit `null` passed. Even though SQL `NULL` cannot be compared against (`WHEN NULL` will never match), it is actually a valid case value.

People would need to be aware of that behavior, and use the `value()` method in case they dynamically pass a value that could be `null`.

---

When would you need to switch on null?

---

I'm not sure that you ever actually _need_ to, but I also don't really know why it's valid on SQL level in the first place, maybe just because failing hard when a column or function return value is `NULL` would not be nice, or maybe there is actually use case for that behavior.

On PHP level it could easily happen with dynamic values. We could forbid `null` for the value, not sure which behavior people would prefer.

---

You can use func_num_args() if NULL support is desired.

---

Oh boy, there's a function that I haven't used in 20 years, I pretty much forgot it even existed. Well OK, then I don't think I see any technical hurdles.